src/storage/storageController.ts

```diff
@@ -63,17 +66,35 @@ export default class StorageController {
     return Promise.resolve();
   }
 
-  getUserID(): string {
-    let globalUserId = this.get(StorageVariables.GLOBAL_USER_ID);
+  getUserIdentity() {
+    return {
+      userId: this.getUserId(),
+      anonymousId: this.getAnonymousId()
+    };
+  }
+
+  getAnonymousId() {
+    let globalAnonymousId = this.get(StorageVariables.GLOBAL_ANONYMOUS_ID);
+
+    if (globalAnonymousId && typeof globalAnonymousId === 'string') {
+      return globalAnonymousId;
+    }
+
+    const globalUserId = this.get(StorageVariables.GLOBAL_USER_ID);
 
     if (globalUserId && typeof globalUserId === 'string') {
-      return globalUserId;
+      globalAnonymousId = globalUserId;
+    } else {
+      globalAnonymousId = uuidv4();
```

Comment:
What's the difference between this and what was here before? Isn't the global user id the same as a global anonymous id then?

Reply:
We should not use userId field anymore since it is not AUID. But to not break the statistic for old users we should keep userId for previously sent telemetry. So the logic now is:
- check if userId exists
- if userId exists, copy userId to anonymousID and send both to segment
- if userId does not exist, generate anonymousID and sent only anonymousID to segment

Reply:
How does the anonymousID differ from the userId if they both use `uuidv4` and are only generated once?

Reply:
We should send the same uuid but as anonymousId instead of userId.